source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,369 @@
+
+While most packages in Clear Linux are typically packaged using 
+`autospec`_, the kernel is not. This means control files provided 
+by autospec are not available and changes must be made manually.
+
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+
+#. Modify the Version, Release, and Source0 URL entries at the top of the 
+   file to change the version of Linux kernel that will be compiled.
+
+   A list of current and available kernel release can be found on 
+   `kernel.org`_.
+
+   .. code-block:: bash
+      
+      Name:           linux
+      Version:        4.20.5
+      Release:        690
+      License:        GPL-2.0
+      Summary:        The Linux kernel
+      Url:            http://www.kernel.org/
+      Group:          kernel
+      Source0:        https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.20.5.tar.xz
+      Source1:        config
+      Source2:        cmdline
+
+   .. note::
+      Consider changing the Name in the RPM spec file to easily identify a modified kernel.
+
+#. Commit and save the changes to the file.
+
+
+
+Modify kernel configuration 
+===========================
+
+Existing kernel features and in-tree kernel modules can be enabled or 
+disabled in the kernel configuration file, :file:`.config` , at compile time.
+
+To manage unique changes that have been made to the kernel config file, 
+|CL| uses a kernel config fragment file named :file:`config-fragment`. 
+Managing kernel configuration changes with a configuration fragment file 
+instead of directly editing the :file:`.config` file helps identify the 
+unique configuration changes that have been made and makes applying any 
+future default configuration values easier. 
+
+The :file:`config-fragment` is the **only** file that is modified and is 
+eventually merged with the main :file:`.config`.
+
+
+
+#. Open the kernel :file:`config-fragment` file in an editor.
+   
+   Due to how |CL| packaging tools make use of :command:`mock` environments 
+   psuedo-GUI tools that abstract kernel configuration such as menuconfig do 
+   not work. 
+
+   .. code-block:: bash
+
+      $EDITOR config-fragment
+
+
+#. Find the configuration values you are looking for. 
+   If a particular setting does not already exist, it can be added manually.
+
+   For example, the snippet below shows BTRFS support configuration indicating
+   it is enabled in-tree.
+
+   .. code-block:: bash
+
+      CONFIG_BTRFS_FS=y
+      CONFIG_BTRFS_FS_POSIX_ACL=y
+
+
+#. Modify the configuration values as desired. 
+
+   For example, the snippet below shows BTRFS support configuration changed 
+   change to be disabled and commented out.
+
+   .. code-block:: bash
+      
+      # CONFIG_BTRFS_FS is not set
+      # CONFIG_BTRFS_FS_POSIX_ACL is not set
+
+#. Commit and save the changes to the :file:`config-fragment` file.
+
+
+#. Run the :command:`make config` command to apply the changes made in the 
```

Comment:
May want to note that this action pulls in configuration fragments from the `kernel-config` package, and sandwiches the options in `config-fragment` between `base-X` and `mandatory-X`

Reply:
I've revamped this section to use a more standard kernel config editing workflow and not force the reader to use the CL-way workflow. 

I think the CL team's method of using kernel fragments is still useful for others to know. Perhaps we can publish a blog post on the subject and reference it here in the future.